addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -0,0 +1,207 @@
+                case CHANNEL_WELCOME_EVENT_CAMERAID:
+                    Optional<AbstractNetatmoThingHandler> camera = getBridgeHandler()
+                            .findNAThing(lastEvent.getCameraId());
+                    return camera.map(c -> toStringType(c.getThing().getLabel())).orElse(UnDefType.UNDEF);
+                case CHANNEL_WELCOME_EVENT_PERSONID:
+                    Optional<AbstractNetatmoThingHandler> person = getBridgeHandler()
+                            .findNAThing(lastEvent.getPersonId());
+                    return person.map(p -> toStringType(p.getThing().getLabel())).orElse(UnDefType.UNDEF);
+                case CHANNEL_WELCOME_EVENT_SNAPSHOT:
+                    String url = getSnapshotURL(lastEvent.getSnapshot());
+                    return url != null ? HttpUtil.downloadImage(url) : UnDefType.UNDEF;
+                case CHANNEL_WELCOME_EVENT_SNAPSHOT_URL:
+                    String snapshotURL = getSnapshotURL(lastEvent.getSnapshot());
+                    return toStringType(snapshotURL);
+                case CHANNEL_WELCOME_EVENT_VIDEO_URL:
+                    if (lastEvent.getVideoId() != null) {
+                        String cameraId = lastEvent.getCameraId();
+                        Optional<AbstractNetatmoThingHandler> thing = getBridgeHandler().findNAThing(cameraId);
+                        if (thing.isPresent()) {
+                            NAWelcomeCameraHandler eventCamera = (NAWelcomeCameraHandler) thing.get();
+                            String streamUrl = eventCamera.getStreamURL(lastEvent.getVideoId());
+                            if (streamUrl != null) {
+                                return new StringType(streamUrl);
+                            }
+                        }
+                    }
+                    return UnDefType.UNDEF;
+                case CHANNEL_WELCOME_EVENT_VIDEOSTATUS:
+                    return toStringType(lastEvent.getVideoStatus());
+                case CHANNEL_WELCOME_EVENT_ISARRIVAL:
+                    return lastEvent.getIsArrival() != null ? OnOffType.ON : OnOffType.OFF;
+                case CHANNEL_WELCOME_EVENT_MESSAGE:
+                    return lastEvent.getMessage() != null
+                            ? new StringType(lastEvent.getMessage().replace("<b>", "").replace("</b>", ""))
+                            : UnDefType.UNDEF;
+                case CHANNEL_WELCOME_EVENT_SUBTYPE:
+                    return toDecimalType(lastEvent.getSubType());
+            }
+        }
+        return super.getNAThingProperty(channelId);
+
+    }
+
+    /**
+     * Returns the Url of the picture
+     *
+     * @return Url of the picture or null
+     */
+    protected String getSnapshotURL(NAWelcomeSnapshot snapshot) {
+        String result = null;
+
+        if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
+            StringBuffer sb = new StringBuffer();
```

Comment:
Just as some background: I you do not do such things in a loop this has no added performance, but no problem in leaving it as is.